src/Controls/src/Core/Shell/BaseShellItem.cs

```diff
@@ -423,9 +423,27 @@ internal static DataTemplate CreateDefaultFlyoutItemCell(string textBinding, str
 				columnDefinitions.Add(new ColumnDefinition { Width = GridLength.Star });
 				defaultGridClass.Setters.Add(new Setter { Property = Grid.ColumnDefinitionsProperty, Value = columnDefinitions });
 
-				Binding automationIdBinding = new Binding(Element.AutomationIdProperty.PropertyName);
+				BindingBase automationIdBinding = TypedBindingFactory.Create(static (Element element) => element.AutomationId);
 				defaultGridClass.Setters.Add(new Setter { Property = Element.AutomationIdProperty, Value = automationIdBinding });
 
+				BindingBase imageBinding = null;
+				BindingBase labelBinding = null;
+
+				if (bo is MenuItem)
+				{
+					imageBinding = TypedBindingFactory.Create(static (MenuItem item) => item.IconImageSource);
+					labelBinding = TypedBindingFactory.Create(static (MenuItem item) => item.Text);
+				}
+				else if (bo is BaseShellItem)
+				{
+					imageBinding = TypedBindingFactory.Create(static (BaseShellItem item) => item.FlyoutIcon);
+					labelBinding = TypedBindingFactory.Create(static (BaseShellItem item) => item.Title);
+				}
+				else
+				{
+					throw new InvalidOperationException("Invalid type for flyout item cell.");
+				}
```

Comment:
Are there any other types that need to work here besides `MenuItem` and `BaseShellItem`? What was the exception you would have gotten previously for using the wrong type inside a flyout?

Or did previously you got no error, and the bindings just "didn't work"?